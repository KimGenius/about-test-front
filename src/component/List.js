import {Table, TableBody, TableCell, TableHead, TableRow} from "@mui/material"
import dayjs from "dayjs"

function List({babList}) {
  return (
    <div>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>신규입사자</TableCell>
            <TableCell align="right">같이 밥 먹은 팀원</TableCell>
            <TableCell align="right">출근한 팀원들</TableCell>
            <TableCell align="right">밥 먹은 날</TableCell>
          </TableRow>
        </TableHead>
        <TableBody data-testid={"list"}>
          {babList.map((bab) => (
            <TableRow
              className={'table-row'}
              key={bab.idx}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {bab.newMember}
              </TableCell>
              <TableCell align="right">{bab.resultMember}</TableCell>
              <TableCell align="right">{bab.targetMember}</TableCell>
              <TableCell align="right">{dayjs(bab.createdAt).format('YYYY-MM-DD HH:mm:ss')}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  )
}

export default List
