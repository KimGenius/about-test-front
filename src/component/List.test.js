import {render, screen, waitFor} from '@testing-library/react'
import List from './List'
import axios from 'axios'

jest.mock('axios')

const data = [
  {
    "idx": 21,
    "newMember": "김영재",
    "targetMember": "강민철, 권윤하, 김민재, 이민우, 이세한, 이지선, 이형주, 최호정, 황순영",
    "createdAt": "2021-10-27T15:17:32.000Z",
    "resultMember": "이지선,이민우"
  },
  {
    "idx": 20,
    "newMember": "김영재",
    "targetMember": "강민철, 권윤하, 김민재, 이민우, 이세한, 이지선, 이형주, 최호정, 황순영",
    "createdAt": "2021-10-27T15:17:31.000Z",
    "resultMember": "이형주,김민재"
  }
]
test('List render 테스트', async () => {
  const { getByTestId, asFragment }= render(<List babList={data} />)
  const listNode = await waitFor(() => getByTestId("list"))
  expect(listNode.children).toHaveLength(data.length)
  expect(asFragment()).toMatchSnapshot()
  let targetEle = screen.getByText('같이 밥 먹은 팀원')
  expect(targetEle).toBeInTheDocument()
  targetEle = screen.getByText('신규입사자')
  expect(targetEle).toBeInTheDocument()
  targetEle = screen.getByText('밥 먹은 날')
  expect(targetEle).toBeInTheDocument()
  targetEle = screen.getAllByText('김영재')
  expect(targetEle).toHaveLength(2)
  targetEle = screen.getAllByText("강민철, 권윤하, 김민재, 이민우, 이세한, 이지선, 이형주, 최호정, 황순영")
  expect(targetEle).toHaveLength(2)
})
