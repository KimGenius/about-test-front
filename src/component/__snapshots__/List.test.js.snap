// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render 테스트 1`] = `
<DocumentFragment>
  <div>
    <table
      aria-label="simple table"
      class="MuiTable-root css-quj9j6-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            신규입사자
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            같이 밥 먹은 팀원
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            출근한 팀원들
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            밥 먹은 날
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        data-testid="list"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            김영재
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            이지선,이민우
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            강민철, 권윤하, 김민재, 이민우, 이세한, 이지선, 이형주, 최호정, 황순영
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2021-10-28 00:17:32
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            김영재
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            이형주,김민재
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            강민철, 권윤하, 김민재, 이민우, 이세한, 이지선, 이형주, 최호정, 황순영
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2021-10-28 00:17:31
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
